From 03bf456cb5b65ff7cfabfab3758ca8f3c6816556 Mon Sep 17 00:00:00 2001
From: Cyril Hrubis <chrubis@suse.cz>
Date: Thu, 4 Sep 2014 11:28:55 +0200
Subject: [PATCH] realtime: Fix exp10 check.

The parameter to exp10() function was constant and because of that it
was substituted by a constant at the compile time.

This patch passes volatile float as the parameter and also adds -lm to
the LIBS which is needed to link the test.

Reported-by: Waldemar Brodkorb <wbx@openadk.org>
Signed-off-by: Cyril Hrubis <chrubis@suse.cz>
---
 testcases/realtime/m4/ltp-exp10.m4 | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/testcases/realtime/m4/ltp-exp10.m4 b/testcases/realtime/m4/ltp-exp10.m4
index 4fe7091..84a3d9b 100644
--- a/testcases/realtime/m4/ltp-exp10.m4
+++ b/testcases/realtime/m4/ltp-exp10.m4
@@ -24,10 +24,14 @@ AC_DEFUN([LTP_CHECK_EXP10],[
 AH_TEMPLATE(HAVE_EXP10,
 [Define to 1 if you have 'exp10' function.])
 AC_MSG_CHECKING([for exp10])
+backup_ldlibs="$LIBS"
+LIBS+=" -lm"
 AC_TRY_LINK([#define _GNU_SOURCE
              #include <math.h>],
             [
-             exp10(0.00);
+	     volatile float val;
+             exp10(val);
             ],
              AC_DEFINE(HAVE_EXP10) AC_MSG_RESULT(yes), AC_MSG_RESULT(no))
+LIBS="$backup_ldlibs"
 ])
-- 
1.8.5.5

